marathon_quiz.cpp

```diff
@@ -452,4 +453,40 @@ void deletePlayer(Player** head){
 
     // If no player with the given ID was found
     printf("No player with ID %i found.\n", playerId);
+}
+
+
+void changeName(Player* head) {
+    int playerId;
+    const char* newNickname;
+    char* nickname;
+   //checks if theres already players
+    if (head == NULL) {
+        printf("There's no player to rename\n");
+        return;
+    }
+
+    Player* current = head;
+	//catchs the id 
+    printf("Gimme the id of the one who wants a fresh new name\n ");
+    scanf("%d", &playerId);
+
+    //looks for the id in the list until its empty or finds the name
+    while (current != NULL && current->id != playerId) {
+        current = current->next;
+    }
+
+    //if the id doesnt exist exits
+    if (current == NULL) {
+        printf("There's no such a player with the ID: %d\n", playerId);
+        return;
+    }
+    //if the id exist now it asks you the name
+    printf("How wants this person to be called?:\n ");
+    scanf(" %[^\n]", nickname);
```

Comment:
looks good I just would advise to replace this line with the nicknameCreation function, but it isn't necessary